its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/CodeCoverageTest.java

```diff
@@ -0,0 +1,127 @@
+  }
+
+  @Test
+  void whenRunningOnAzureDevops_coverageIsImported() throws IOException {
+    var projectDir = TestUtils.projectDir(basePath, PROJECT_NAME);
+    var token = TestUtils.getNewToken(ORCHESTRATOR);
+
+    List<EnvironmentVariable> environmentVariables = VstsUtils.isRunningUnderVsts()
+      ? Collections.emptyList()
+      : // In order to simulate Azure-DevOps environment, the following variables are needed:
+        // TF_Build -> Must be set to true.
+        // BUILD_BUILDURI -> Must have a value. The value can be anything.
+        // AGENT_BUILDDIRECTORY -> The agent build directory; the tests results should be present in a child "TestResults" folder.
+        Arrays.asList(
+          new EnvironmentVariable("TF_Build", "true"),
+          new EnvironmentVariable("BUILD_BUILDURI", "fake-uri"),
+          new EnvironmentVariable("AGENT_BUILDDIRECTORY", projectDir.toString()));
+
+    runBeginStep(projectDir, token, environmentVariables);
+    runTestsWithCoverage(projectDir);
+
+    var endStepResult = runEndStep(projectDir, token, environmentVariables);
+    assertThat(endStepResult.getLogs()).contains("Coverage report conversion completed successfully.");
+    assertThat(endStepResult.getLogs()).containsPattern("Parsing the Visual Studio coverage XML report .*coveragexml");
+    assertThat(endStepResult.getLogs()).contains("Coverage Report Statistics: 2 files, 1 main files, 1 main files with coverage, 1 test files, 0 project excluded files, 0 other language files.");
+  }
+
+  private static void runBeginStep(Path projectDir, String token, List<EnvironmentVariable> environmentVariables) {
+      ORCHESTRATOR.getServer().provisionProject(PROJECT_KEY, PROJECT_NAME);
+      var scanner = TestUtils.newScanner(ORCHESTRATOR, projectDir, token)
+        .addArgument("begin")
+        .setProjectKey(PROJECT_KEY)
+        .setProjectName(PROJECT_NAME)
+        .setProperty("sonar.projectBaseDir", projectDir.toAbsolutePath().toString())
+        .setProperty("sonar.verbose", "true")
+        .setProjectVersion("1.0");
+      for (var environmentVariable : environmentVariables) {
+        scanner.setEnvironmentVariable(environmentVariable.getName(), environmentVariable.getValue());
+      }
+      var beginStepResult = ORCHESTRATOR.executeBuild(scanner);
+      assertTrue(beginStepResult.isSuccess());
+  }
+
+  private static void runTestsWithCoverage(Path projectDir) {
+    // On AzureDevops the build directory is already set, and it's different from the "projectDir"
+    // In order to simulate the behavior, we need to generate the test results in the location specified by %AGENT_BUILDDIRECTORY%
+    var buildDirectory = VstsUtils.getEnvBuildDirectory() == null ? projectDir.toString() : VstsUtils.getEnvBuildDirectory();
+    var testResult = TestUtils.runDotnetCommand(projectDir, "test", "--collect", "Code Coverage", "--logger", "trx", "--results-directory", buildDirectory + "\\TestResults");
```

Comment:
Add a comment that `--collect "Code Coverage"` produces a binary coverage file `.coverage` that needs to be converted to an XML `.coveragexml` file by the end step.